compiler-rt/test/sanitizer_common/TestCases/memcmp.cpp

```diff
@@ -0,0 +1,15 @@
+// RUN: %clangxx -O0 %s -o %t && %run %t
+// XFAIL: *
```

Comment:
I don't think this is correct. You want ` && not %run %t` above. Probably add some FileCheck to assert it actually fails with a HWASan error, not a segfault.

Reply:
I guess it's not likely to segfault, but the allocator could in principle return the page on delete.

Reply:
I guess for interceptors like these optimal testing is:

1. sanitizer_common - basic positive test check that it links and DO NOT CRASH, maybe basic functionality test
2. in each sanitizer - report test

  AddressSanitizer-i386-linux :: TestCases/Posix/bcmp_test.cpp
  AddressSanitizer-i386-linux :: TestCases/memcmp_strict_test.cpp
  AddressSanitizer-i386-linux :: TestCases/memcmp_test.cpp

  MemorySanitizer-X86_64 :: memcmp_test.cpp

So I suggest to adopt for HWASAN:
  AddressSanitizer-i386-linux :: TestCases/Posix/bcmp_test.cpp
  AddressSanitizer-i386-linux :: TestCases/memcmp_strict_test.cpp
  AddressSanitizer-i386-linux :: TestCases/memcmp_test.cpp

Reply:
Positive test for memcmp probably unnecessary as it's quite commonly used, and we should notice if it's obviously broken.